libsplinter/src/oauth/mod.rs

```diff
@@ -177,6 +178,20 @@ impl OAuthClient {
         Ok(Some((user_info, pending_authorization.client_redirect_url)))
     }
 
+    /// Exchanges the given refresh token for an access token.
+    pub fn exchange_refresh_token(&self, refresh_token: String) -> Result<String, InternalError> {
+        self.client
+            .exchange_refresh_token(&RefreshToken::new(refresh_token))
+            .request(http_client)
+            .map(|response| response.access_token().secret().into())
+            .map_err(|err| {
+                InternalError::with_message(format!(
```

Comment:
This should probably be from_source_with_message(Box::new(err), "Failed to make refresh token exchange request".to_string)

Reply:
Unfortunately the errors returned by the `oauth2` crate don't implement `std::error::Error` so they can't be used as a source with `InternalError`.